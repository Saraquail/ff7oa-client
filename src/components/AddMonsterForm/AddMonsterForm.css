

#add-monster-form button {
  margin-bottom: 1em;
  margin-top: 0;
  margin-left: 2em;
}

input {

  width: 10em;
  background-color: #696969;
  color: whitesmoke;
  border: solid 2px #555454;
  box-shadow: #5a5350;
  font-size: 16px;
  transition-property: background;
  transition-duration: .5s;
  transition-timing-function: linear;
}

input:focus {
  background-color: #f1b1c5;
  border: solid 1px #807e80;
  color:  rgb(90, 35, 67);
}


.form-description {
  margin: auto;
  margin-bottom: 1em;
  width: fit-content;
  background-image: linear-gradient(to bottom, rgb(128, 47, 94), rgb(90, 35, 67));
  padding: .25em;
  border-radius: .5em;
  border: solid 1px #424542;
  box-shadow: 1px 2px #e7dfe7,
              -1px -2px #e7dfe7,
              /* 1px -1px #e7dfe7,
              -1px 1px #e7dfe7, */
              -2px 0 #7b757b,
              0 2px #424542;
}

.all-the-inputs { 
  display: grid;
  grid-gap: .5em;
  grid-template-columns: repeat(auto-fit, minmax(350px, 1fr)); 
}

.input-w-label {
  text-align: right;
  margin-right: 15%;
  margin-bottom: 1em;
}

label {
  margin-right: .5em;
}

@media (min-width: 600px) {
  .all-the-inputs { 
    margin-right: 1em;
    margin-left: 1em;
  }
}

@media (min-width: 900px) {
  .all-the-inputs { 
    text-align: right;
    margin: auto
  }
}