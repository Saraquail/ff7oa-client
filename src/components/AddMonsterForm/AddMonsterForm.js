import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import MonsterApiService from '../../services/monster-api-service'
import TokenService from '../../services/token-service'
import './AddMonsterForm.css'

class AddMonsterForm extends Component {

  state = {
    name: '',
    hp: '',
    mp: '',
    exp: '',
    gil: '',
    weakness: 'N/A',
    strength: 'N/A',
    location: '',
    level: '',
    steal: 'N/A',
    drops: 'N/A',
    enemy_skill: 'N/A',
    message: ''
  }

  handleInputChange = (ev) => {
    ev.preventDefault()

    const target = ev.target;
    const value = target.value 
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleCancelForm = () => {
    this.props.history.goBack()
  }

  handleSubmitForm = ev => {
    ev.preventDefault()
    let monster = this.state
    let user = TokenService.getUserName()
    this.validateMonster(user, monster)
    }

    validateMonster = (user, monster) => {
      let { hp, mp, gil, exp, level }  = this.state
      
      if (hp < 1 || hp > 1000000) {
        this.setState({
          message: 'HP must be between 1 - 1,000,000'
        })
      }
      else if (mp < 1 || mp > 200000) {
        this.setState({
          message: 'MP must be between 1 - 200,000'
        })
      }
      else if (gil < 1 || mp > 200000) {
        this.setState({
          message: 'GIL must be between 1 - 200,000'
        })
      }
      else if (exp < 1 || exp > 500000) {
        this.setState({
          message: 'EXP must be between 1 - 500,000'
        })
      }
      else if (level < 1 || level > 99) {
        this.setState({
          message: 'Level must be between 1 - 99'
        })
      }
      else {
        MonsterApiService.postMonster(user, monster)
        .then(res => res.id ? this.props.history.goBack()
        : '')
        .catch(e => this.setState({ message: e.error }))
      }
    }

  render () {
    return (
      <div className="add-monster-container">
      <h1>Add a new monster </h1>
        <form name="add-monster-form" id="add-monster-form" onSubmit={this.handleSubmitForm}>
        <div className='form-description'>
          All fields are required. If monster has no weakness/strength/drop/steal, simply put N/A as the value for that field.
        </div>
          <button type="reset" onClick={this.handleCancelForm}>Not Interested</button>
          <label htmlFor="name">Name</label>
          <input type="text" name="name" id="name" required onChange={this.handleInputChange} />
          <label htmlFor="hp">HP (number)</label>
          <input type="number" name="hp" id="hp" required onChange={this.handleInputChange} />
          <label htmlFor="mp">MP (number)</label>
          <input type="number" name="mp" id="mp" required onChange={this.handleInputChange} />
          <label htmlFor="mp">EXP (number)</label>
          <input type="number" name="exp" id="exp" required onChange={this.handleInputChange} />
          <label htmlFor="gil">GIL (number)</label>
          <input type="number" name="gil" id="gil" required onChange={this.handleInputChange} />
          <label htmlFor="gil">Weakness</label>
          <input type="text" name="weakness" id="weakness" defaultValue = 'N/A' required onChange={this.handleInputChange} />
          <label htmlFor="strength">Strength</label>
          <input type="text" name="strength" id="strength" defaultValue = 'N/A' required onChange={this.handleInputChange} />
          <label htmlFor="location">Location</label>
          <input type="text" name="location" id="location" required onChange={this.handleInputChange} />
          <label htmlFor="level">Level (number)</label>
          <input type="number" name="level" id="level" required onChange={this.handleInputChange} />
          <label htmlFor="steal">Steal</label>
          <input type="text" name="steal" id="steal" defaultValue = 'N/A' required onChange={this.handleInputChange} />
          <label htmlFor="drops">Drops</label>
          <input type="text" name="drops" id="drops" defaultValue = 'N/A' 
          required onChange={this.handleInputChange} />
          <label htmlFor="enemy_skill">Enemy Skill</label>
          <input type="text" name="enemy_skill" id="enemy_skill" defaultValue = 'N/A' required onChange={this.handleInputChange} />
          <p id="message">{this.state.message}</p>
          <button type="submit">OK, add it</button>
          <button type="reset" onClick={this.handleCancelForm}>Not Interested</button>
        </form>
      </div>
    )
  }
}

export default withRouter(AddMonsterForm)

