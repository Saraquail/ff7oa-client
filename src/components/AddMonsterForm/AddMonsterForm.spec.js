import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import AddMonsterForm from './AddMonsterForm.js';

describe('<AddMonsterForm/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <AddMonsterForm />
      </BrowserRouter>, div,
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AddMonsterForm />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
