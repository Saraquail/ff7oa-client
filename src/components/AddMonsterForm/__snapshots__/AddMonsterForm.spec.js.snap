// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddMonsterForm/> renders the UI as expected 1`] = `
<div
  className="add-monster-container"
>
  <h1>
    Add a new monster 
  </h1>
  <form
    id="add-monster-form"
    name="add-monster-form"
    onSubmit={[Function]}
  >
    <div
      className="form-description"
    >
      All fields are required. If monster has no weakness/strength/drop/steal, simply put N/A as the value for that field.
    </div>
    <p
      className="form-description"
    >
       When you add a monster, the field "Added by" will be added to your monster with your username.
    </p>
    <button
      onClick={[Function]}
      type="reset"
    >
      Not Interested
    </button>
    <div
      className="all-the-inputs"
    >
      <div
        className="input-w-label"
      >
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          autoComplete="none"
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="hp"
        >
          HP
        </label>
        <input
          autoComplete="none"
          id="hp"
          max={1000000}
          min={0}
          name="hp"
          onChange={[Function]}
          required={true}
          type="number"
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="mp"
        >
          MP
        </label>
        <input
          autoComplete="none"
          id="mp"
          max={20000}
          min={0}
          name="mp"
          onChange={[Function]}
          required={true}
          type="number"
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="exp"
        >
          EXP
        </label>
        <input
          autoComplete="none"
          id="exp"
          max={500000}
          min={0}
          name="exp"
          onChange={[Function]}
          required={true}
          type="number"
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="gil"
        >
          GIL
        </label>
        <input
          autoComplete="none"
          id="gil"
          max={200000}
          min={0}
          name="gil"
          onChange={[Function]}
          required={true}
          type="number"
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="strength"
        >
          Strength
        </label>
        <input
          autoComplete="none"
          defaultValue="N/A"
          id="strength"
          name="strength"
          onChange={[Function]}
          required={true}
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="weakness"
        >
          Weakness
        </label>
        <input
          autoComplete="none"
          defaultValue="N/A"
          id="weakness"
          name="weakness"
          onChange={[Function]}
          required={true}
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="location"
        >
          Location
        </label>
        <input
          autoComplete="none"
          id="location"
          name="location"
          onChange={[Function]}
          required={true}
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="level"
        >
          Level
        </label>
        <input
          autoComplete="none"
          id="level"
          max={99}
          min={0}
          name="level"
          onChange={[Function]}
          required={true}
          type="number"
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="drops"
        >
          Drops
        </label>
        <input
          autoComplete="none"
          defaultValue="N/A"
          id="drops"
          name="drops"
          onChange={[Function]}
          required={true}
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="steal"
        >
          Steal
        </label>
        <input
          autoComplete="none"
          defaultValue="N/A"
          id="steal"
          name="steal"
          onChange={[Function]}
          required={true}
        />
      </div>
      <div
        className="input-w-label"
      >
        <label
          htmlFor="enemy_skill"
        >
          Enemy Skill
        </label>
        <input
          autoComplete="none"
          defaultValue="N/A"
          id="enemy_skill"
          name="enemy_skill"
          onChange={[Function]}
          required={true}
        />
      </div>
    </div>
    
    <button
      type="submit"
    >
      OK, add it
    </button>
    <button
      onClick={[Function]}
      type="reset"
    >
      Not Interested
    </button>
  </form>
</div>
`;
